src/test/com/epam/lab/payments/MainRestControllerTest.java

```diff
@@ -0,0 +1,74 @@
+package com.epam.lab.payments;
+
+
+import com.epam.lab.payments.dao.UserRepository;
+import com.epam.lab.payments.domain.UserEntity;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = Payments.class)
+@WebAppConfiguration
+public class MainRestControllerTest {
+
+
+    private MediaType contentType = new MediaType(MediaType.APPLICATION_JSON.getType(),
+            MediaType.APPLICATION_JSON.getSubtype(),
+            Charset.forName("utf8"));
+
+    private MockMvc mockMvc;
+
+    private String userEmail = "bdussault";
+
+    private String password = "12345";
+
+    private HttpMessageConverter mappingJackson2HttpMessageConverter;
+
+    private UserEntity userEntity;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    void setConverters(HttpMessageConverter<?>[] converters) {
+
+        this.mappingJackson2HttpMessageConverter = Arrays.asList(converters).stream().filter(
+                hmc -> hmc instanceof MappingJackson2HttpMessageConverter).findAny().get();
```

Comment:
You can write like this Arrays.stream(converters)..
Try not to use .get() on Optional without check. This will fail by NPE if there won't be value inside Optional.
And update formatting because now it is quite unreadable.

Reply:
Rework to orElse and reformat